### ISSUE TITLE
Speed up dsm.var.gam() and dsm.var.prop() by replacing numderiv() with grad() from numDeriv package

### ISSUE THREAD

**Issue by jjrob** [DistanceDevelopment/dsm#6]
dsm.var.gam() and dsm.var.prop() call sapply to compute derivatives on potentially large matrices using the numderiv() helper function, e.g.:

```R
dtmfn <- function(eta){sapply(eta, numderiv, f=tmfn)}
...
dpred.db[ipg,] <- this.off.set %**% (dtmfn(lppred)*lpmat)
```

Instead, consider using grad() from the numDeriv package, replacing the first line above with something like:

```R
dtmfn <- function(eta){grad(tmfn, eta)}
```
In a test of a 300,000 element lppred vector, the runtime on my machine dropped from about 5 minutes to 24 seconds.

**Comment by jjrob** [comment]
Note that grad() from the numDeriv package does not handle NAs gracefully:

```R
> dtmfn <- function(eta){grad(tmfn, eta)}
> dtmfn(c(1, 2, 3))
[1]  2.718282  7.389056 20.085537
> dtmfn(c(1, 2, 3, NA))
Error in grad.default(tmfn, eta) : 
  function returns NA at 1e-042e-043e-04NA distance from x.
```

You could work around with something like this:

```R
> dtmfn <- function(eta){ifelse(is.na(eta), NA, grad(tmfn, ifelse(is.na(eta), 0, eta)))}
> dtmfn(c(1, 2, 3, NA))
[1]  2.718282  7.389056 20.085537        NA
```

So that if you put NAs in, you'd get NAs out...

**Comment by dill** [comment]
Implemented in a8d2492. Thanks @jjrob !
